src/cli/init.ts

```diff
@@ -101,12 +101,20 @@ export const initCommand = new Command()
         default: "groth16",
         choices: [{ name: "Groth16", value: "groth16" }],
       });
-      const curveName: "bn254" = await select({
+      const curveName = await select({
         message: "Curve Name:",
         default: "bn254",
-        choices: [{ name: "BN254", value: "bn254" }],
+        choices: [
+          { name: "BN254", value: "bn254" },
+          { name: "BLS12_377", value: "bls12-377" },
+          { name: "BLS12_381", value: "bls12-381" },
+          { name: "BLS24_315", value: "bls24-315" },
+          { name: "BW6_633", value: "bw6-633" },
+          { name: "BW6_761", value: "bw6-761" },
+        ],
       });
-      Object.assign(context, { curveName, packageName, provingScheme });
+      const gnarkCurveName = curveName.toUpperCase().replace("-", "_");
```

Comment:
Some cleanup needs to happen here to get stronger typing, something that we'll want as I add tests around the templating.

Which is the format we want to display, and which to send to the API? Looking at

```python
class GnarkCurveOptions(CaseInsensitiveStringEnum):
    # https://docs.gnark.consensys.net/Concepts/schemes_curves#docusaurus_skipToContent_fallback
    bls12_377 = "bls12-377"
    bls12_381 = "bls12-381"
    bls24_315 = "bls24-315"
    bn254 = "bn254"
    bw6_633 = "bw6-633"
    bw6_761 = "bw6-761"
```

it seems like kebab case is the canonical value? This also seems more readable to me for enum literals.

Reply:
This is a bit nuanced. Kebab case is what we send to the API via the `sindri.json` field value `curve`.  And the I/O part of the init command formats the options like that.  The line you have commented here becomes necessary because of the `FromJson` function of our circuit template.  Specifically we need to invoke a gnark-crypto structure via its struct name, not via a string.  See https://github.com/Sindri-Labs/sindri-js/blob/9f68cfaf30ee3b9c94a01dfcca54fa2096e85a72/templates/gnark/templatePackageName.go#L65

Reply:
Thanks, makes sense.